contracts/0.8.25/vaults/dashboard/Dashboard.sol

```diff
@@ -216,11 +216,36 @@ contract Dashboard is NodeOperatorFee {
 
     /**
      * @notice Disconnects the staking vault from the vault hub.
+     * VaultHub stores data for calculating the node operator fee, so the fees should be claimed first.
      */
     function voluntaryDisconnect() external {
+        if (nodeOperatorUnclaimedFee() > 0) revert NodeOperatorFeeUnclaimed();
+
         _voluntaryDisconnect();
     }
 
+    /**
+     * @notice Accepts the ownership over the staking vault transferred from VaultHub on disconnect
+     * and immediately transfers it to a new pending owner. This new owner will have to accept the ownership
+     * on the staking vault contract.
+     * @param _newOwner The address to transfer the staking vault ownership to.
+     */
+    function abandonDashboard(address _newOwner) external {
+        address vaultAddress = address(_stakingVault());
+        if (VAULT_HUB.vaultSocket(vaultAddress).vault == vaultAddress) revert ConnectedToVaultHub();
+
+        _acceptOwnership();
+        _transferOwnership(_newOwner);
+    }
+
+    /**
+     * @notice Reclaims the Dashboard contract, transferring the staking vault ownership to the VaultHub.
+     * @dev This function is used when the Dashboard contract is no longer needed and the staking vault should be transferred to a new owner.
+     */
+    function connectToVaultHub() external {
```

Comment:
Don't we need to acceptOwnership first? After disconnection owner is still vaultHub, but pendingOwner is Dashboard.

Reply:
fixed